test/rpc_test.py

```diff
@@ -159,55 +163,58 @@ def test_register_rpc_backend_and_init_rpc_backend(
         "PROCESS_GROUP rpc backend specific test, skip",
     )
     def test_duplicate_name(self):
-        store = dist.FileStore(self.file_name, self.world_size)
+        init_method = (
+            "file://{file_name}?rank={rank}&world_size={world_size}"
+        ).format(
+            file_name=self.file_name, rank=self.rank, world_size=self.world_size
+        )
+
         dist.init_process_group(
-            backend="gloo", rank=self.rank, world_size=self.world_size, store=store
+            backend="gloo", init_method=init_method
         )
         with self.assertRaisesRegex(RuntimeError, "is not unique"):
             dist.init_model_parallel(
                 self_name="duplicate_name",
                 backend=BACKEND,
                 self_rank=self.rank,
-                init_method=RPC_INIT_URL,
+                init_method=init_method,
             )
         dist.join_rpc()
 
+    @_wrap_with_rpc
     def test_reinit(self):
-        store = dist.FileStore(self.file_name, self.world_size)
-        dist.init_process_group(
-            backend="gloo", rank=self.rank, world_size=self.world_size, store=store
-        )
-        dist.init_model_parallel(
-            self_name="worker{}".format(self.rank),
-            backend=BACKEND,
-            self_rank=self.rank,
-            init_method=RPC_INIT_URL,
-        )
+        # dist.init_model_parallel(...) called in `@_wrap_with_rpc` first.
         with self.assertRaisesRegex(RuntimeError, "is already initialized"):
             dist.init_model_parallel(
                 self_name="worker{}".format(self.rank),
                 backend=BACKEND,
                 self_rank=self.rank,
-                init_method=RPC_INIT_URL,
+                init_method=self.init_method,
```

Comment:
where is self.init_method? could it be set in setup()?

Reply:
@zhaojuanmao 

If I set it in `def setUp`, `def _wrap_with_rpc` will not be able to access it.

Reply:
@zhaojuanmao 

It's set in `def _wrap_with_rpc`.

No we can't put per-worker setup in `def setUp()`, because `def setUp` only runs on parent process.

Reply:
you can set self.init_method for each child processing in __run() class method, same as self.rank, self.file_name

Reply:
you can also set it as class property, just like word_size?

Reply:
@zhaojuanmao 

I don't quite follow the "class property" idea.
Are you suggesting adding this?
```
@property
def init_method(self):
  pass
```

Reply:
@zhaojuanmao 

Thanks for the idea.

Isn't `def _run` used by c10d distributed tests as well?

Will I contaminate it by setting up a `self.init_method` it won't use?

Reply:
In summary, I think there is no clean way to do it, if we don't do a big refactoring to current test code.

The thing needs to be done is to make every worker run `setUp()`, and put RPC set up logic in this `setUp()` method, rather than calling a decorator on every test method manually.